### PR TITLE
Feature/dashboard direct debit and insurance status

### PR DESCRIPTION
**Pull request by FredrikAreschoug**

### CODE REVIEW

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt [hunk 1]
```diff
@@ -28,6 +46,15 @@ class DashboardFragment : Fragment() {
 
     private var personalCoverageCardOpen: Boolean = false
 
+    private var isInsurancePendingExplanationExpanded = false
+
+    private var setActivationFiguresInterval: Disposable? = null
+    private val compositeDisposable = CompositeDisposable()
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt]: Should we try out AutoDispose? 🙂

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt [hunk 2]
```diff
@@ -62,6 +101,8 @@ class DashboardFragment : Fragment() {
                     "NAME" to d.member().firstName()
                 )
                 setupLargeTitle(title, R.font.circular_bold)
+                setupDirectDebitStatus(data.directDebitStatus())
+                setupInsuranceStatusStatus(data.insurance().status(), data.insurance().activeFrom())
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt]: Not super-important but we could probably act directly on Data.Insurance here as we access multiple properties from it. Bike-shedding though 🤷‍♂️

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt [hunk 3]
```diff
@@ -105,6 +146,97 @@ class DashboardFragment : Fragment() {
...
+            InsuranceStatus.INACTIVE_WITH_START_DATE -> {
+                insurancePending.show()
+                activeFrom?.let { localDate ->
+                    insurancePendingCountDownContainer.show()
+                    insurancePendingLoadingAnimation.remove()
+
+                    setActivationFigures(localDate)
+                    val formattedString = localDate.format(DateTimeFormatter.ofPattern("d LLLL yyyy"))
+                    insurancePendingExplanation.text = interpolateTextKey(getString(R.string.DASHBOARD_DIRECT_DEBIT_STATUS_PENDING_HAS_START_DATE_EXPLANATION), "START_DATE" to formattedString)
+                } ?: run {
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt]: I've considered doing this as well but I feel like it reads a bit poorly. 🤷‍♂️

File: owldroid/src/main/java/com/hedvig/android/owldroid/util/extensions/view/AnimationExtensions.kt [hunk 4]
```diff
@@ -5,13 +5,20 @@ import android.animation.ValueAnimator
...
 import android.view.animation.DecelerateInterpolator
+import android.widget.TextView
 
 fun View.animateExpand(
     duration: Long = 200,
     interpolator: TimeInterpolator = DecelerateInterpolator()
 ) {
-    measure(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
-    val targetHeight = measuredHeight
+    val targetHeight = if (this is TextView) {
```
Comment (hedvigoscar) [inline owldroid/src/main/java/com/hedvig/android/owldroid/util/extensions/view/AnimationExtensions.kt]: 🙊

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt (hunk 1 above)
Comment (FredrikAreschoug) [reply owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt]: Texh depth :+1:

File: owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt (hunk 2 above)
Comment (FredrikAreschoug) [reply owldroid/src/main/java/com/hedvig/android/owldroid/ui/dashboard/DashboardFragment.kt]: :+1:

**Pull request merged**